lib/services/unified_resource.go

```diff
@@ -741,122 +741,133 @@ const (
+		//nolint:staticcheck // SA1019. TODO(sshah) DELETE IN 17.0
+		switch appOrSP := resource.(type) {
+		case *types.AppServerV3:
+			protoResource = &proto.PaginatedResource{
+				Resource: &proto.PaginatedResource_AppServerOrSAMLIdPServiceProvider{
+					AppServerOrSAMLIdPServiceProvider: &types.AppServerOrSAMLIdPServiceProviderV1{
+						Resource: &types.AppServerOrSAMLIdPServiceProviderV1_AppServer{
+							AppServer: appOrSP,
 						},
-					}}
-			case *types.SAMLIdPServiceProviderV1:
-				protoResource = &proto.PaginatedResource{
-					Resource: &proto.PaginatedResource_AppServerOrSAMLIdPServiceProvider{
-						AppServerOrSAMLIdPServiceProvider: &types.AppServerOrSAMLIdPServiceProviderV1{
-							Resource: &types.AppServerOrSAMLIdPServiceProviderV1_SAMLIdPServiceProvider{
-								SAMLIdPServiceProvider: appOrSP,
-							},
+					},
+				}, RequiresRequest: requiresRequest}
+		case *types.SAMLIdPServiceProviderV1:
+			protoResource = &proto.PaginatedResource{
+				Resource: &proto.PaginatedResource_AppServerOrSAMLIdPServiceProvider{
+					AppServerOrSAMLIdPServiceProvider: &types.AppServerOrSAMLIdPServiceProviderV1{
+						Resource: &types.AppServerOrSAMLIdPServiceProviderV1_SAMLIdPServiceProvider{
+							SAMLIdPServiceProvider: appOrSP,
 						},
-					}}
-			default:
-				return nil, trace.BadParameter("%s has invalid type %T", resourceKind, resource)
-			}
-		case types.KindSAMLIdPServiceProvider:
-			serviceProvider, ok := resource.(*types.SAMLIdPServiceProviderV1)
-			if !ok {
-				return nil, trace.BadParameter("%s has invalid type %T", resourceKind, resource)
-			}
-
-			protoResource = &proto.PaginatedResource{Resource: &proto.PaginatedResource_SAMLIdPServiceProvider{SAMLIdPServiceProvider: serviceProvider}}
+					},
+				}, RequiresRequest: requiresRequest}
 		default:
-			return nil, trace.NotImplemented("resource type %s doesn't support pagination", resource.GetKind())
+			return nil, trace.BadParameter("%s has invalid type %T", resourceKind, resource)
+		}
+	case types.KindSAMLIdPServiceProvider:
+		serviceProvider, ok := resource.(*types.SAMLIdPServiceProviderV1)
+		if !ok {
+			return nil, trace.BadParameter("%s has invalid type %T", resourceKind, resource)
 		}
 
+		protoResource = &proto.PaginatedResource{Resource: &proto.PaginatedResource_SAMLIdPServiceProvider{SAMLIdPServiceProvider: serviceProvider}, RequiresRequest: requiresRequest}
+	default:
+		return nil, trace.NotImplemented("resource type %s doesn't support pagination", resource.GetKind())
+	}
+
+	return protoResource, nil
+}
+
+// MakePaginatedResources converts a list of resources into a list of paginated proto representations.
+func MakePaginatedResources(requestType string, resources []types.ResourceWithLabels, requestableMap map[string]bool) ([]*proto.PaginatedResource, error) {
+	paginatedResources := make([]*proto.PaginatedResource, 0, len(resources))
+	for _, r := range resources {
+
+		_, requiresRequest := requestableMap[r.GetName()]
```

Comment:
Might as well use `map[string]struct{}` over `map[string]bool` if you're going to do the check this way, saves some bytes and there's no ambiguity of what a `false` in the map would mean

Reply:
Oh thats cool, didn't know you could do that. thank you!